pkg/api/validation/events.go

```diff
@@ -25,11 +25,20 @@ import (
 // ValidateEvent makes sure that the event makes sense.
 func ValidateEvent(event *api.Event) errs.ValidationErrorList {
 	allErrs := errs.ValidationErrorList{}
-	if event.Namespace != event.InvolvedObject.Namespace {
-		allErrs = append(allErrs, errs.NewFieldInvalid("involvedObject.namespace", event.InvolvedObject.Namespace, "namespace does not match involvedObject"))
-	}
+
 	if !util.IsDNSSubdomain(event.Namespace) {
 		allErrs = append(allErrs, errs.NewFieldInvalid("namespace", event.Namespace, ""))
 	}
+	// TODO: event.InvolvedObject is not versioned. References need to be normalizd to internal when converted from external and vice versa.
+	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind == "Node" {
```

Comment:
@derekwaynecarr didn't you mention that you needed (or someone else was going to) fix the namespace defaulting for events?

----- Original Message -----
> >  	if !util.IsDNSSubdomain(event.Namespace) {
> >  		allErrs = append(allErrs, errs.NewFieldInvalid("namespace",
> >  		event.Namespace, ""))
> >  	}
> > +	// TODO: event.InvolvedObject is not versioned. References need to be
> > normalizd to internal when converted from external and vice versa.
> > +	if event.InvolvedObject.Kind == "Minion" || event.InvolvedObject.Kind ==
> > "Node" {
> 
> Our api code changed a lot after I created the PR.  I'll need to
> re-investigate to see if it works.  Previously, it won't work because event
> is not root namespaced, so api server will silently change its None
> namespace to Default, which violates the validation.
> 
> 
> 
> 
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/4396/files#r24950442